### ISSUE TITLE
expose hostNetwork pod setting in values.yaml

### ISSUE THREAD

**Issue by jrhunger** [signalfx/splunk-otel-collector-chart#849]
### Is your feature request related to a problem? Please describe.

We're only using the agent to scrape container logs, so we don't have a requirement to listen on host ports for connections from other pods.   Currently hostNetwork: true is hardcoded [in the chart ](https://github.com/signalfx/splunk-otel-collector-chart/blob/6d3ca53d1e70f60a9c166217dd509a8a376cc058/helm-charts/splunk-otel-collector/templates/daemonset.yaml#L52) if not on Windows.

### Describe the solution you'd like

Expose an option via --set / values.yaml for disabling use of hostNetwork.

### Describe alternatives you've considered

Tried using the [ports section](https://github.com/signalfx/splunk-otel-collector-chart/blob/main/helm-charts/splunk-otel-collector/values.yaml#L318-L361) to disable the ports by setting to null:

```
agent:
  # disable daemon ports since only scraping logs
  ports:
    otlp: null
    otlp-http: null
    otlp-http-old: null
    fluentforward: null
```

Agent still listened on the ports
```
lsof -nP -p 4167 | grep -i tcp
otelcol 4167 root    3u     IPv4 103201689       0t0       TCP 127.0.0.1:55554 (LISTEN)
otelcol 4167 root    7u     IPv6 103200229       0t0       TCP *:8889 (LISTEN)
otelcol 4167 root    8u     IPv6 103201696       0t0       TCP *:13133 (LISTEN)
otelcol 4167 root    9u     IPv4 103201699       0t0       TCP 127.0.0.1:55679 (LISTEN)
otelcol 4167 root   10u     IPv4 103199430       0t0       TCP 10.75.144.66:36036->172.16.128.1:443 (ESTABLISHED)
otelcol 4167 root   11u     IPv6 103202255       0t0       TCP *:8006 (LISTEN)
otelcol 4167 root   13u     IPv6 103201702       0t0       TCP *:4317 (LISTEN)
otelcol 4167 root   14u     IPv6 103201703       0t0       TCP *:4318 (LISTEN)
```

As a workaround, pulled the helm chart and patched the template before applying.

**Comment by atoulme** [comment]
I have filed an idea on your behalf here: https://ideas.splunk.com/ideas/EID-I-2101

Please upvote and flag to your Splunk rep.

**Comment by kumachop2** [comment]
@jrhunger Please do share the workaround applied to the template if you're don't mind. Much appreciated your response.

**Comment by jrhunger** [comment]
Sure. Here is what we do:
```
# pull helm repo
helm repo add splunk "$HELM_REPO"
helm repo update splunk
/bin/rm -rf splunk-otel-collector
helm pull --version $CHART_VERSION splunk/splunk-otel-collector --untar

# patch hostNetwork to false on daemonset
sed -i 's/hostNetwork: true/hostNetwork: false/' splunk-otel-collector/templates/daemonset.yaml
```

**Comment by kumachop2** [comment]
@jrhunger Thank You for your prompt response. Actually l'm looking for workaround to remediate "Usage of host networking and ports should be restricted" defender defect and setting hostNetwork  to false did resolved, but otel port need to be enabled for APM traces.

**Comment by theBNT** [comment]
Btw as mentioned in issue #1014 this has been implemented with PR #1107 and released with [0.92.0](https://github.com/signalfx/splunk-otel-collector-chart/releases/tag/splunk-otel-collector-0.92.0)
